pkg/epp/datastore/datastore.go

```diff
@@ -227,68 +269,48 @@ func (ds *datastore) ModelGetAll() []*v1alpha2.InferenceModel {
 }
 
 // /// Pods/endpoints APIs ///
-func (ds *datastore) PodUpdateMetricsIfExist(namespacedName types.NamespacedName, m *Metrics) bool {
-	if val, ok := ds.pods.Load(namespacedName); ok {
-		existing := val.(*PodMetrics)
-		existing.Metrics = *m
-		return true
-	}
-	return false
-}
 
-func (ds *datastore) PodGet(namespacedName types.NamespacedName) *PodMetrics {
-	val, ok := ds.pods.Load(namespacedName)
-	if ok {
-		return val.(*PodMetrics)
-	}
-	return nil
+func (ds *datastore) PodGetAll() []backendmetrics.PodMetrics {
+	return ds.PodList(func(backendmetrics.PodMetrics) bool { return true })
 }
 
-func (ds *datastore) PodGetAll() []*PodMetrics {
-	res := []*PodMetrics{}
+func (ds *datastore) PodList(predicate func(backendmetrics.PodMetrics) bool) []backendmetrics.PodMetrics {
+	res := []backendmetrics.PodMetrics{}
 	fn := func(k, v any) bool {
-		res = append(res, v.(*PodMetrics))
+		pm := v.(backendmetrics.PodMetrics)
+		if predicate(pm) {
+			res = append(res, pm)
+		}
 		return true
 	}
 	ds.pods.Range(fn)
 	return res
 }
 
-func (ds *datastore) PodRange(f func(key, value any) bool) {
-	ds.pods.Range(f)
-}
-
-func (ds *datastore) PodDelete(namespacedName types.NamespacedName) {
-	ds.pods.Delete(namespacedName)
-}
-
-func (ds *datastore) PodUpdateOrAddIfNotExist(pod *corev1.Pod) bool {
-	new := &PodMetrics{
-		Pod: Pod{
-			NamespacedName: types.NamespacedName{
-				Name:      pod.Name,
-				Namespace: pod.Namespace,
-			},
-			Address: pod.Status.PodIP,
-		},
-		Metrics: Metrics{
-			ActiveModels: make(map[string]int),
+func (ds *datastore) PodUpdateOrAddIfNotExist(pod *corev1.Pod, pool *v1alpha2.InferencePool) bool {
+	bp := &backendmetrics.Pod{
```

Comment:
how about we pass `corev1.Pod` to NewPodMetrics, looks more appropriate than creating half of PodMetrics struct here and the other half in NewPodMetrics.